gtdynamics/kinematics/KinematicsSlice.cpp

```diff
@@ -105,11 +137,22 @@ Values Kinematics::initialValues<Slice>(const Slice& slice, const Robot& robot,
 
 template <>
 Values Kinematics::inverse<Slice>(const Slice& slice, const Robot& robot,
-                                  const ContactGoals& contact_goals) const {
-  auto graph = this->graph(slice, robot);
+                                  const ContactGoals& contact_goals,
+                                  bool contact_goals_as_constraints) const {
 
-  // Add objectives.
-  graph.add(pointGoalObjectives(slice, contact_goals));
+  // Robot kinematics constraints
+  auto constraints = this->constraints(slice, robot);
+  NonlinearFactorGraph graph;
+
+  // Contact goals
+  if (contact_goals_as_constraints) {
+    constraints.add(this->pointGoalConstraints(slice, contact_goals));
```

Comment:
What is `EqualityConstraints` anyway? I thought we had decided on just using a vector?

Reply:
I discovered the need of `emplace_shared` and adding all equalityconstraints from another `EqualityConstraint` class. (same as member functions of `FactorGraph`)